12-project-maintenance.Rmd

```diff
@@ -8,3 +8,121 @@ ottrpal::set_knitr_image_path()
+
+In order to keep maintenance a priority without it using up all your time, we have a few tips to discuss. These tips will help you ease the burden of maintenance and lead to a healthier R project long term.
+
+Building some of the infrastructure we will describe will take a bit more time upfront, but will save you time long term.
+
+1. Have unit testing with `testthat`
+2. Use continuous deployment / continuous integration (CI/CD) principles
+
+## What is Unit Testing?
+
+One of the most comprehensive methods for making sure your code works is building tests. Unit testing can be really straightforward to build using the [`testthat` package](https://testthat.r-lib.org/). `testhat` works with the `usethis` R package so you can start by installing both of those.
+
+```
+install.packages(c("usethis", "testthat"))
+```
+
+Then you can create your first test:
+
+```
+usethis::use_test("new_test")
+```
+
+This will open up a page that looks like this:
+
+```
+test_that("multiplication works", {
+  expect_equal(2 * 2, 4)
+})
+```
+
+From this chapter:
+
+> - A test file holds one or more test_that() tests.
+> - Each test describes what it’s testing: e.g. “multiplication works”.
+> - Each test has one or more expectations: e.g. expect_equal(2 * 2, 4).
+
+A key to unit testing is having a proper test for *every piece of functionality you have written*.
+
+To read more about the details of testing we [recommend this book chapter](https://r-pkgs.org/testing-basics.html).
+
+You can run all your tests using `devtools::test()` or by setting up handy automation that will run it for you. We will discuss that now!
+
+## What is CI/CD?
+
+At the core of CI/CD is using automation to boost the reproducibility of your work!
+
+Robots are much better at repetitive work. In other words, your human collaborator is great at many things but even your most reliable collaborator will not be as punctual as a robot who is programmed to do the job.
+
+Let's bring this into the terms of a very common story for science. Let's say you are a researcher who submitted a manuscript and a reviewer comes back and asks you to re run the analysis with a minor tweak; perhaps a parameter change.
+
+```{r, out.width = "100%", echo = FALSE}
+ottrpal::include_slide("https://docs.google.com/presentation/d/1MNHf8JpolaEP_vQ_kB-1xRBF9wo3haCArRu117hBoHA/edit#slide=id.g33bf0789107_300_812")
+```
+
+If you developed your analysis without using reproducibility aiding practices and without automation, it is very likely that this seemingly simple task could take a lot of your time and brain power. Because while you don't think anything on your computer changed since you ran this analysis 6 months ago, your computing environment and the software it uses has been changing the entire time!
```

Comment:
```suggestion
If you developed your analysis without using reproducibility aiding practices and without automation, it is very likely that this seemingly simple task could take a lot of your time and brain power. While you might not think anything on your computer changed since you ran this analysis 6 months ago, your computing environment and the software it uses has been changing the entire time!
```

Reply:
some might know